src/probnum/filtsmooth/statespace/discrete_transition.py

```diff
@@ -131,64 +86,43 @@ class DiscreteLinearGaussian(DiscreteGaussian):
-            return dynamatfct(t, **kwargs) @ x + forcefct(t, **kwargs)
+    def __init__(
+        self,
+        dynamicsmatfun: Callable[[FloatArgType], np.ndarray],
+        forcevecfun: Callable[[FloatArgType], np.ndarray],
+        diffmatfun: Callable[[FloatArgType], np.ndarray],
+    ):
 
-        def jacfct(t, x, **kwargs):
-            return dynamatfct(t, **kwargs)
+        self.dynamicsmatfun = dynamicsmatfun
+        self.forcevecfun = forcevecfun
 
-        super().__init__(dynafct, diffmatfct, jacfct)
-        self._forcefct = forcefct
+        super().__init__(
+            dynamicsfun=lambda t, x: (self.dynamicsmatfun(t) @ x + self.forcevecfun(t)),
+            diffmatfun=diffmatfun,
+            jacobfun=lambda t, x: dynamicsmatfun(t),
+        )
+
+    def transition_rv(self, rv, start, **kwargs):
+        # **kwargs swallow all irrelevant arguments for this function.
 
-    def transition_rv(self, rv, start, stop=None, **kwargs):
         if not isinstance(rv, pnrv.Normal):
             raise TypeError(f"Normal RV expected, but {type(rv)} received.")
-        dynamat = self.dynamicsmatrix(time=start)
-        diffmat = self.diffusionmatrix(time=start)
-        force = self.forcevector(time=start)
 
-        new_mean = dynamat @ rv.mean + force
-        new_crosscov = rv.cov @ dynamat.T
-        new_cov = dynamat @ new_crosscov + diffmat
-        return pnrv.Normal(mean=new_mean, cov=new_cov), {"crosscov": new_crosscov}
+        dynamicsmat = self.dynamicsmatfun(start)
+        diffmat = self.diffmatfun(start)
+        force = self.forcevecfun(start)
 
-    def dynamicsmatrix(self, time, **kwargs):
-        """
-        Compute dynamics matrix :math:`G=G(t)` at time :math:`t`.
-        The output is equivalent to :meth:`jacobian`.
-
-        Parameters
-        ----------
-        time : float
-            Time :math:`t`.
-
-        Returns
-        -------
-        :class:`np.ndarray`
-            Evaluation of the dynamics matrix :math:`G=G(t)`.
-        """
-        return self._jacobfun(time, None, **kwargs)
-
-    def forcevector(self, time, **kwargs):
-        """
-        Compute force vector :math:`v=v(t)` at time :math:`t`.
-
-        Parameters
-        ----------
-        time : float
-            Time :math:`t`.
-
-        Returns
-        -------
-        :class:`np.ndarray`
-            Evaluation of the force :math:`v=v(t)`.
-        """
-        return self._forcefct(time, **kwargs)
+        new_mean = dynamicsmat @ rv.mean + force
+        new_crosscov = rv.cov @ dynamicsmat.T
+        new_cov = dynamicsmat @ new_crosscov + diffmat
+        return pnrv.Normal(mean=new_mean, cov=new_cov), {"crosscov": new_crosscov}
 
     @property
     def dimension(self):
-        return len(self.dynamicsmatrix(0.0).T)
+        # risky to evaluate at zero, but works well
+        # remove this -- the dimension of discrete transitions is not clear!
+        # input dim != output dim is possible...
```

Comment:
these comments show how unsure I am about this property... it works (for now), but seems like a very dangerous few lines of code to me...

Reply:
In this case I agree that the comments are useful! Maybe also open an issue to complement this?